TileStache/Mapnik.py

```diff
@@ -74,19 +74,24 @@ class ImageProvider:
         - http://trac.mapnik.org/wiki/XMLConfigReference
     """
 
-    def __init__(self, layer, mapfile, fonts=None, scale_factor=None):
-        """ Initialize Mapnik provider with layer and mapfile.
+    def __init__(self, layer, mapconfig, fonts=None, scale_factor=None):
+        """ Initialize Mapnik provider with layer and mapconfig.
 
-            XML mapfile keyword arg comes from TileStache config,
+            XML mapconfig keyword arg comes from TileStache config,
             and is an absolute path by the time it gets here.
         """
-        maphref = urljoin(layer.config.dirpath, mapfile)
+        maphref = urljoin(layer.config.dirpath, mapconfig)
```

Comment:
> I don't like how a non-path is being passed to this function. We need to detect that it is not file differently somehow

any suggestions? My idea was to look at the URI which I would is sufficient. The other option would be to try to open the file and if it fails then assume it is a xml string. But then a invalid path would be treated as xml string too.